### ISSUE TITLE
Allow passing of keyword arguments

### ISSUE THREAD

**Issue by christianscott** [christianscott/curry.py#3]
This will require some pretty big changes to the way the function works. Will need to seperate the `signature(fun).parameters` list into two: args and kwargs.

The kwargs could be stored in a dict, then could check if the dict has enough keys of the right type before calling the function.

Not sure how this would interact with default values.

Something like this:
```python
def curried(*args, **kwargs):
    assert only_one_has_value(args, kwargs)
    if kwargs:
       # set value
    if len(arguments) == arg_count or all_keys_set(kwarguments, needed_keys):
        ...
```

**Comment by praveensvsrk** [comment]
There's a small bug in the code. The content of the list args is persistent over various calls. Is this intentional?
![2017-09-28](https://user-images.githubusercontent.com/10628197/30948763-d953fbc4-a42f-11e7-807f-c947977412b1.png)

**Comment by praveensvsrk** [comment]
Why not `func(3, b=5)` ?

**Comment by christianscott** [comment]
Not quite sure what you mean! Would you mind explaining? I was talking about something like this:

```python
add(a=1)(b=2)
```
Currently this produces an error

**Comment by praveensvsrk** [comment]
Assertion `assert only_one_has_value(args, kwargs)`  fails when someone does this: `add(3, b=4)`. But I feel this should work as normal?

**Comment by christianscott** [comment]
Oh I see. Yeah you're right, that's probably way too strict. Thanks for helping me out by the way!

**Comment by christianscott** [comment]
fixed in 07e9f0a2894472977d420054c09dd65123076f8b
